src/java.base/linux/native/libnio/ch/FileDispatcherImpl.c

```diff
@@ -71,6 +71,8 @@ Java_sun_nio_ch_FileDispatcherImpl_transferFrom0(JNIEnv *env, jobject this,
         if (errno == EINTR) {
             return IOS_INTERRUPTED;
         }
+        if (errno == EPERM)
+            return IOS_UNSUPPORTED;
```

Comment:
This is unrelated to [JDK-8316304](https://bugs.openjdk.org/browse/JDK-8316304). If you want to fix this as well, then please as a separate patch. The bug speaks about JDK-8316304 and JDK-8316304 only.